string_filters.hpp

```diff
@@ -124,6 +133,23 @@ inline constexpr lowercase_t lowercase{};
  */
 struct canonicalize_t {
     std::string process(const std::string& str) const {
+#ifdef WITHOUT_VIEWS
+        /** FIXME: gcc 11 doesn't like the range solution, ergo this hack */
+        std::vector<std::string> words;
+        std::istringstream f(str);
+        std::string s;
+        while (std::getline(f, s, ' ')) {
+            words.push_back(s);
```

Comment:
To go with the return suggestion I made below, this should probably be words.push_back(s | capitalize)